ert_shared/ensemble_evaluator/entity/ensemble.py

```diff
@@ -7,60 +12,124 @@
 )
 from res.enkf import EnKFState
 
+from pathlib import Path
+from collections import defaultdict
+from graphlib import TopologicalSorter
+
 
 logger = logging.getLogger(__name__)
 
 
+def _sort_steps(steps: typing.List["_Step"]):
+    graph = defaultdict(set)
+    if len(steps) == 1:
+        return (steps[0].get_name(),)
+    edged_nodes = set()
+    for step in steps:
+        for other in steps:
+            if step == other:
+                continue
+            step_outputs = set([io.get_name() for io in step.get_outputs()])
+            other_inputs = set([io.get_name() for io in other.get_inputs()])
+            if len(step_outputs) > 0 and step_outputs.issubset(other_inputs):
```

Comment:
Should we not add the edge from `other` to `step` if `step` produce any output that `other` needs? In other words, if `step_outputs` and `other_inputs` is not disjoint? If so, the tests should also be improved...